tensorflow/compiler/mlir/tensorflow/transforms/optimize.td

```diff
@@ -17,8 +17,9 @@ include "mlir/Dialect/StandardOps/IR/Ops.td"
 include "tensorflow/compiler/mlir/tensorflow/ir/tf_ops.td"
 
 def IsDataFormatNHWC : ConstantAttr<TF_ConvnetDataFormatAttr, "NHWC">;
-def BroadcastableElements :
-    Constraint<CPred<"TFL::IsBroadcastableElementsAttrs($0, $1)">>;
+def CanFuseMulAndConv2D :
+    Constraint<CPred<"TFL::IsBroadcastableElementsAttrs($0, $1) && TFL::CanFuseAffineOpThenMul($1)">>;
```

Comment:
why mul->conv2d needs a condition of `affine->mul`? can you rename CanFuseAffineOpThenMul to a more general name?